ical.js

```diff
@@ -192,92 +219,95 @@
   return {
 
 
-    objectHandlers : {
-      'BEGIN' : function(component, params, curr, stack){
-          stack.push(curr)
+    objectHandlers: {
+      'BEGIN': function(component, params, curr, stack) {
+        stack.push(curr)
 
-          return {type:component, params:params}
+        return {
+          type: component,
+          params: params
         }
+      }
 
-      , 'END' : function(component, params, curr, stack){
+      ,
+      'END': function(component, params, curr, stack) {
         // prevents the need to search the root of the tree for the VCALENDAR object
         if (component === "VCALENDAR") {
-            //scan all high level object in curr and drop all strings
-            var key,
-                obj;
-            
-            for (key in curr) {
-                if(curr.hasOwnProperty(key)) {
-                   obj = curr[key];
-                   if (typeof obj === 'string') {
-                       delete curr[key];
-                   }
-                }
+          //scan all high level object in curr and drop all strings
+          var key,
+            obj;
+
+          for (key in curr) {
+            if (curr.hasOwnProperty(key)) {
+              obj = curr[key];
+              if (typeof obj === 'string') {
+                delete curr[key];
+              }
             }
-            
-            return curr
+          }
+
+          return curr
         }
-        
-        var par = stack.pop()
 
-        if (curr.uid)
-          par[curr.uid] = curr
-        else
-          par[Math.random()*100000] = curr  // Randomly assign ID : TODO - use true GUID
+        var par = stack.pop()
+        par[UUID.v4()] = curr;
 
         return par
       }
 
-      , 'SUMMARY' : storeParam('summary')
-      , 'DESCRIPTION' : storeParam('description')
-      , 'URL' : storeParam('url')
-      , 'UID' : storeParam('uid')
-      , 'LOCATION' : storeParam('location')
-      , 'DTSTART' : dateParam('start')
-      , 'DTEND' : dateParam('end')
-      ,' CLASS' : storeParam('class')
-      , 'TRANSP' : storeParam('transparency')
-      , 'GEO' : geoParam('geo')
-      , 'PERCENT-COMPLETE': storeParam('completion')
-      , 'COMPLETED': dateParam('completed')
-      , 'CATEGORIES': categoriesParam('categories')
-      , 'FREEBUSY': freebusyParam('freebusy')
+      ,
+      'SUMMARY': storeParam('summary'),
+      'DESCRIPTION': storeParam('description'),
+      'URL': storeParam('url'),
+      'UID': storeParam('uid'),
+      'LOCATION': storeParam('location'),
+      'DTSTART': dateParam('start'),
+      'DTEND': dateParam('end'),
+      ' CLASS': storeParam('class'),
```

Comment:
I checked the spec and there is no reason to believe that this has whitespace before it. This is definitely a typo